falcon/media/json.py

```diff
@@ -12,19 +12,61 @@ class JSONHandler(BaseHandler):
 
     This handler uses Python's :py:mod:`json` by default, but will
     use :py:mod:`ujson` if available.
+
+    Keyword Arguments:
+        dumps (func): the function to use when serializing JSON responses
+
+        loads (func): the function to use when deserializing JSON requests
+
+        dumps_kwargs (dict): keyword arguments passed to ``dumps`` when serializing to JSON.
+            ``ensure_ascii`` is set to ``false`` by default. To override this setting, pass ``True``
+            explicitly.
+
+        loads_kwargs (func): keyword arguments to pass to ``loads`` when deserializing from JSON.
+
+
+    You can override the JSON library used by changing the ``dumps`` and ``loads`` functions::
+
+        import falcon
+        from falcon import media
+        import rapidjson
+
+
+        json_handler = media.JSONHandler(
+            dumps=rapidjson.dumps,
+            loads=rapidjson.loads,
+            dumps_kwargs=dict(ensure_ascii=False)
+        )
+        extra_handlers = {
+            'application/json': json_handler,
+            'application/json; charset=UTF-8': json_handler,
+        }
+
+        api = falcon.API()
+        api.req_options.media_handlers.update(extra_handlers)
+        api.resp_options.media_handlers.update(extra_handlers)
     """
 
+    def __init__(self, dumps=None, dumps_kwargs=None, loads=None, loads_kwargs=None):
+        self.dumps = dumps or json.dumps
+        self.dumps_kwargs = dict(ensure_ascii=False)
+        self.dumps_kwargs.update(dumps_kwargs or dict())
```

Comment:
Could this set the self.dumps/loads_kwargs arguments to None if not explicitly set, and call self.dumps/loads without **kwargs if kwargs is None? This would support orjson, which doesn't take kwargs.

Reply:
In a pinch, one could wrap ``orjson.dumps`` in a kwargs-discarding closure, and pass that as the *dumps* parameter, but I agree it would be cleaner to have a way to set empty *kwargs*.